projects/plugins/jetpack/extensions/blocks/ai-image/edit.js

```diff
@@ -54,13 +54,17 @@ function getImagesFromOpenAI(
 			} );
 			setResultImages( images );
 		} )
-		.catch( () => {
-			setErrorMessage(
-				__(
-					'Whoops, we have encountered an error. AI is like really, really hard and this is an experimental feature. Please try again later.',
-					'jetpack'
-				)
-			);
+		.catch( e => {
+			if ( e.data?.user_message ) {
```

Comment:
Frankly I think we should just use the regular `message` payload instead of introducing a new format

Reply:
Taht would be more consistent with the WP_Error format instead of introducing custom fields

Reply:
I feel you, but the idea was to prevent [those errors](fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Syvo%2Sbcranv%2Sbcranv.cuc%23104-og) to be thrown into the user's face and [get the default error instead](https://github.com/Automattic/jetpack/blob/1cf7d4b0c478ed1287599e1e05390b56af6a65c8/projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js#L223)...

Reply:
But I can take it the other way around as well and fit the WP_Error format. I'll work on that quickly.

Reply:
So on backend we could just catch the openai errors and throw a different one with the user-facing error message

Reply:
YEs yes.